### PR TITLE
O2-Czech-Republic#244 allow specifying consistency level

### PR DESCRIPTION
**Pull request by je-ik**
Closes O2-Czech-Republic#244.

### CODE REVIEW

**Comment by je-ik** [comment]
@tisonet can you have a look at this, please?

File: direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java [hunk 1]
```diff
@@ -84,39 +84,57 @@
...
         log.warn("Failed to instantiate type converter {}", tmp, ex);
       }
     }
-    this.converter = (StringConverter) c;
-    initializeCqlFactory();
+    return c;
+  }
+
+  private ConsistencyLevel getConsistencyLevel(Map<String, Object> cfg) {
+    return Optional.ofNullable(cfg.get("consistency-leve"))
```
Comment (tisonet) [inline direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java:134]: typo, chybi `l` v level

File: direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java [hunk 2]
```diff
@@ -84,39 +84,57 @@
...
+        .map(Object::toString)
+        .map(ConsistencyLevel::valueOf)
+        .orElse(ConsistencyLevel.QUORUM);
```
Comment (tisonet) [inline direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java:137]: Nechces spise `LOCAL_QUORUM `

File: direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java [hunk 3]
```diff
@@ -84,39 +84,57 @@
   @Nullable private transient Cluster cluster;
   /** Session we are connected to. */
   @Nullable private transient Session session;
+  /** Quorum for both reads and writes. */
```
Comment (tisonet) [inline direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java:87]: Ten komentar nejak nesedi.

Comment (je-ik) [reply direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java:87]: How come?

File: direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java (hunk 1 above)
Comment (je-ik) [reply direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java:134]: thanks

File: direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java (hunk 2 above)
Comment (je-ik) [reply direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java:137]: No, `LOCAL_QUORUM` does not work with replication strategy `SimpleStrategy`. It is meant to be for multi-cluster setups, which should not be the default.

**Review by je-ik — COMMENTED** [review]
Thanks for the review, let's use english, please. The PRs might be read by someone that does not speak czech. Thanks!

File: direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java (hunk 3 above)
Comment (tisonet) [reply direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java:87]: Quorum is single type of more possible levels so this field can hold also another consistency level.

**Review by tisonet — APPROVED** [review]

File: direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java (hunk 3 above)
Comment (je-ik) [reply direct/io-cassandra/src/main/java/cz/o2/proxima/direct/cassandra/CassandraDBAccessor.java:87]: Ah, I think the problem is with the naming of the ConsistencyLevel (QUORUM). Quorum means "the number of people needed to approve a decision". That can be ONE, TWO, MAJORITY (which should be the correct name for QUORUM). So I think that the comment is right, problematic is `ConsistencyLevel.QUORUM` :)

**Pull request merged**